shutdown/legacy.go

```diff
@@ -0,0 +1,84 @@
+package shutdown
+
+import (
+	"github.com/ThomasObenaus/go-base/shutdown/health"
+	"github.com/ThomasObenaus/go-base/shutdown/log"
+	"github.com/ThomasObenaus/go-base/shutdown/signal"
+	"github.com/ThomasObenaus/go-base/shutdown/stop"
+	"github.com/rs/zerolog"
+)
+
+type ShutdownHandler struct {
+	stoppableItems stopIF
+	signalHandler  signalHandlerIF
+	log            logIF
+	health         healthIF
+}
+
+func InstallHandler(orderedStopables []stop.Stoppable, logger zerolog.Logger) *ShutdownHandler {
+	shutdownHandler := &ShutdownHandler{
+		stoppableItems: &stop.OrderedStoppableList{},
+		log:            log.ShutdownLog{Logger: logger},
+		health:         &health.Health{},
+	}
+
+	for _, stopable := range orderedStopables {
+		err := shutdownHandler.stoppableItems.AddToBack(stopable)
```

Comment:
why so complicated / why this indirection

an `handler.orderedStopables = append(handler.orderedStopables, orderedStopables...)` would be enough in my opinion

Reply:
ah. you meant the entire indirection, not this particular instance.

This is the result of the separation of concerns. i have one small unit that handles concurrent access to the stoppableItems. The unit allows for adding, stopping and prevents adding when stopping is in progress.

I could imagine a Solution that does not require this indirection. But I have to try it out to see if it works.